File: provider-quarkus/src/main/java/org/pedrohrr/pacts/provider/MessageResource.java
package org.pedrohrr.pacts.provider;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/messages")
public class MessageResource {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Response hello() {
        return new Response("Hello");
    }
}

class Response {
    private final String message;

    Response(final String message) {
        this.message = message;
    }

    public String getMessage() {
        return message;
    }
}

File: provider-spring/src/main/kotlin/org/pedrohrr/pacts/provider/resources/MessageResource.kt
package org.pedrohrr.pacts.provider.resources

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/messages")
class MessageResource {

    @GetMapping
    fun getMessage(@RequestParam("parameter") parameter: String): Response {
        return Response("any message in here")
    }

}

data class Response(val message: String)